ArduCopter/mode_rc_car.cpp

```diff
@@ -0,0 +1,1000 @@
+    brake.gain = (15.0f * (float)g.poshold_brake_rate + 95.0f) * 0.01f;
+
+    if (copter.ap.land_complete) {
+        // if landed begin in loiter mode
+        roll_mode = RPMode::LOITER;
+        pitch_mode = RPMode::LOITER;
+    } else {
+        // if not landed start in pilot override to avoid hard twitch
+        roll_mode = RPMode::PILOT_OVERRIDE;
+        pitch_mode = RPMode::PILOT_OVERRIDE;
+    }
+
+    // initialise loiter
+    loiter_nav->clear_pilot_desired_acceleration();
+    loiter_nav->init_target();
+
+    // initialise wind_comp each time PosHold is switched on
+    init_wind_comp_estimate();
+
+    flow_filter.set_cutoff_frequency(copter.scheduler.get_loop_rate_hz(), flow_filter_hz);
+
+    quality_filtered = 0;
+    flow_pi_xy.reset_I();
+    limited = false;
+
+    flow_pi_xy.set_dt(1.0/copter.scheduler.get_loop_rate_hz());
+
+    // start with INS height
+    last_ins_height = copter.inertial_nav.get_position_z_up_cm() * 0.01;
+    height_offset = 0;
+
+    return true;
+}
+
+// poshold_run - runs the PosHold controller
+// should be called at 100hz or more
+void ModeRCCar::run()
+{
+    bool rccar_flow_hold = !copter.is_outdoors_ready();
+    if (rccar_flow_hold) {
+        update_height_estimate();
+        if (!is_equal(flow_filter.get_cutoff_freq(), flow_filter_hz)) {
+            flow_filter.set_cutoff_frequency(flow_filter_hz);
+        }
+    }
+    float controller_to_pilot_roll_mix; // mix of controller and pilot controls.  0 = fully last controller controls, 1 = fully pilot controls
+    float controller_to_pilot_pitch_mix;    // mix of controller and pilot controls.  0 = fully last controller controls, 1 = fully pilot controls
+    const Vector3f& vel = inertial_nav.get_velocity_neu_cms();
+
+    // set vertical speed and acceleration limits
+    pos_control->set_max_speed_accel_z(-get_pilot_speed_dn(), g.pilot_speed_up, g.pilot_accel_z);
+    loiter_nav->clear_pilot_desired_acceleration();
+
+    // apply SIMPLE mode transform to pilot inputs
+    update_simple_mode();
+
+    // convert pilot input to lean angles
+    float target_roll, target_pitch;
+    get_pilot_desired_lean_angles(target_roll, target_pitch, copter.aparm.angle_max, attitude_control->get_althold_lean_angle_max_cd());
+
+    float pilot_yaw_rate = get_pilot_desired_yaw_rate(channel_yaw->norm_input_dz());	
+    // rotate roll, pitch input from north facing to vehicle's perspective	
+    // float roll_vel =  vel.y * ahrs.cos_yaw() - vel.x * ahrs.sin_yaw(); // body roll vel	
+    float pitch_vel = vel.y * ahrs.sin_yaw() + vel.x * ahrs.cos_yaw(); // body pitch vel	
+    // gain scheduling for yaw	
+    float pitch_vel2 = MIN(fabsf(pitch_vel), 2000);	
+    float target_yaw_rate = pilot_yaw_rate;
+    // if (fabsf(pitch_vel) > 20 && fabsf(target_pitch) > 1000) { //greater than 20 cm/s
+        target_yaw_rate += (pitch_vel/fabsf(pitch_vel)) * target_roll * (1.0f - (pitch_vel2 / 5000.0f)) * g2.command_model_acro_y.get_rate() / 45.0;	
```

Comment:
RC car mode also includes drift mode code.
This code is slightly  modified to make the behavior as desired